jenkinsbuilder/cccp_index_reader.py

```diff
@@ -100,16 +108,93 @@ def get_projects_from_index(indexdlocation):
                                 dependson, notifyemail, desiredtag)
                         )
                     except Exception as e:
+                        print e
                         raise
     return projects
 
 
+def export_new_project_names(projects_names):
+    """
+    Exports the name of project
+    """
+    # opens the file defined at the start of this script
+    projects_names = "\n".join(projects_names)
+    try:
+        with open(projects_list, "w") as fin:
+            fin.write(projects_names)
+    except IOError as e:
+        print "Failed to export project names to file %s" % projects_list
+        print "I/O Error {0}:{1}".format(e.errno, e.strerror)
+    except:
+        print "Unexpected error:", sys.exc_info()[0]
+    else:
+        # change the mod of file so that jenkins user can edit it
+        print "Changing file permission 0777 of file %s" % projects_list
+        run_command(["chmod", "0777", projects_list])
+
+
+def get_old_project_list():
+    """
+    This function retuns the list of projects that were indexed
+    during last run of cccp-index job. It returns [] empty list
+    if it can not find the project list.
+    """
+    if not os.path.exists(projects_list):
+        print "%s does not exist. This is first run or file is absent." \
+            % projects_list
+        return []
+    with open(projects_list) as fin:
+        old_projects = fin.read()
+    return list(set(old_projects.strip().split("\n")))
+
+
+def find_stale_projects(old, new):
+    """
+    This function diffs the old project list with new project
+    list. And returns the list of project names to be deleted
+    """
+    return list(set(old) - set(new))
+
+
+def delete_stale_projects_on_jenkins(stale_projects):
+    """
+    This function deletes the stale entries at jenkins that
+    are no longer valid. If there is any issue/exception, it
+    bypasses by printing the error.
+    """
+    for project in stale_projects:
+        myargs = ["jenkins-jobs", "delete", project]
+        # print either output or error
+        print run_command(myargs)
```

Comment:
What if there's an error in executing the `run_command`? We don't seem to be doing any exception handling here.